src/tools/MyTools.cpp

```diff
@@ -261,7 +198,7 @@ int drawRandomWithWeights(std::vector<double> weights) {
 	double randNumber = randomDouble(0,sumWeights);
 	int index = 0;
 	double partialSum=weights[0];
-	for (int i=0; i < weights.size()-1; i++) {
+	for (unsigned int i=0; i < weights.size()-1; i++) {
```

Comment:
Do you know why put unsigned int instead of int in every loop?

Reply:
because size() is unsigned. Otherwise we have a compiler warning.